src/oracle/oracle.cairo

```diff
@@ -1055,62 +1061,54 @@ mod Oracle {
         ) -> PossibleEntries {
             let _entry = match data_type {
                 DataType::SpotEntry(pair_id) => {
-                    self.oracle_data_entry_storage.read((pair_id, SPOT, source, 0))
+                    get_entry_storage(self, pair_id, SPOT, source, 0)
                 },
                 DataType::FutureEntry((
                     pair_id, expiration_timestamp
                 )) => {
-                    self
-                        .oracle_data_entry_storage
-                        .read((pair_id, FUTURE, source, expiration_timestamp))
+                    get_entry_storage(self, pair_id, FUTURE, source, expiration_timestamp)
                 },
                 DataType::GenericEntry(key) => {
-                    self.oracle_data_entry_storage.read((key, GENERIC, source, 0))
+                    get_entry_storage(self, key, GENERIC, source, 0)
                 }
             };
-
-            assert(!_entry.is_zero(), 'No data entry found');
+            assert(!_entry.timestamp.is_zero(), 'No data entry found');
```

Comment:
let's change the error message, to `Entry timestamp is null.`

Reply:
The objective of this assertion is to check wether there is or not an entry recorded for the given parameters. If the timestamp is set to 0 for given parameters, it means not only that there are no timestamp set, BUT also, due to architecture, that there are no entry at all recorded for these parameters. Thus, we can let the assertion like that since there are no case where a entry is set without its timestamp.